infection_monkey/monkey.py

```diff
@@ -70,7 +73,16 @@ def start(self):
 
         if firewall.is_enabled():
             firewall.add_firewall_rule()
-        ControlClient.wakeup(parent=self._parent, default_tunnel=self._default_tunnel)
+        ControlClient.find_server(default_tunnel=self._default_tunnel)
+
+        if WindowsUpgrader.should_upgrade():
+            self._upgrading_to_64 = True
+            self._singleton.unlock()
+            LOG.info("32bit monkey running on 64bit Windows. Upgrading.")
+            WindowsUpgrader.upgrade(self._opts)
```

Comment:
What happens if/when we fail to find a server?
Inside upgrade, we'll throw an exception upon trying to open the downloaded monkey file. Then what? 
We're back to whether the Monkey should keep running with no Island service to talk to. Right now, it keeps working and this changes things, by making it fail. I think we better keep running with a 32 bit version and mark in the log file that the upgrade failed.

Reply:
I think the required change is to suicide if couldn't find server.
That's a more responsible behaviour. Maybe later when we see a need for this we'll revert.
I changed it for now so monkey will die